FeedAPIChallenge/Feed API/FeedImageItemMapper.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  FeedImageMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by André Bongartz on 09.09.21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+internal enum FeedImageItemMapper {
+	private struct Root: Decodable {
+		let items: [Item]
+
+		var feedImages: [FeedImage] {
+			return items.map { $0.item }
+		}
+	}
+
+	// decoupling RemoteAPI from internal impelemtion
+	private struct Item: Decodable {
+		var image_id: UUID
+		var image_desc: String?
+		var image_loc: String?
+		var image_url: URL
+
+		var item: FeedImage {
+			return FeedImage(id: image_id, description: image_desc, location: image_loc, url: image_url)
+		}
+	}
+
+	internal static func map(_ data: Data, response: HTTPURLResponse) -> RemoteFeedLoader.Result {
+		guard response.statusCode == 200, let dataAsJson = try? JSONDecoder().decode(Root.self, from: data) else {
```

Comment:
It is good practice to avoid using numbers directly in source code without explanation - by defining a named constant in place of numeric literals we can provide more context on what is represented for the next developer.

```suggestion
		guard response.statusCode == OK_200, let dataAsJson = try? JSONDecoder().decode(Root.self, from: data) else {
```

The `OK_200` property can be a static property as there's no need for the `RemoteFeedLoader` to retrieve a constant value. 

```swift
private static var OK_200: Int { return 200 }
```